docs: update docs for create sink + release notes

ruchirK opened the pull request:
Edit: ready for review!

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/materializeinc/materialize/2550)
<!-- Reviewable:end -->

ruchirK commented on www/layouts/partials/sql-grammar/bnf/create-sink.bnf:
```diff
@@ -1,9 +1,8 @@
-      'SCHEMA'
-         ('FILE' path | inline_schema)
-      )
+   'INTO' (
```
Unsure re: formatting here + the generated diagram is very wide not sure how to fix that

ruchirK replied:
<img width="1153" alt="Screen Shot 2020-04-06 at 12 17 20 PM" src="https://user-images.githubusercontent.com/4015012/78580623-a8782280-7800-11ea-9fa4-50f1dc48619e.png">

ruchirK replied:
@sploiselle maybe you know what to do here? I tried playing around with max width on the website to no avail

sploiselle replied:
This is sadly a limitation of the tool when one arm is very long and one arm is very short. The best solution for this is to do something hacky and break these up into multiple files. I'm glad to either do that for you or talk you through it.

ruchirK replied:
Lets talk through it - I'll message you separately

ruchirK replied:
Done

ruchirK commented on doc/user/sql/create-sink.md:
```diff
@@ -29,19 +29,17 @@ _item&lowbar;name_ | The name of the source or view you want to send to the sink
 **CONFLUENT SCHEMA REGISTRY** _url_ | The URL of the Confluent schema registry to get schema information from.
-**SCHEMA FILE** _path_ | The absolute path to a file containing the schema.
-**SCHEMA** _inline&lowbar;schema_ | A string representing the schema.
+**FILE** _topic_ | The absolute path of the file you want to use as the root of the sink. 
```
not sure how to best convey that these are not the actual file or topic name but rather file / topic name prefixes or "virtual" names

ruchirK commented on doc/user/sql/create-sink.md:
```diff
@@ -29,19 +29,17 @@ _item&lowbar;name_ | The name of the source or view you want to send to the sink
 
-- Materialize currently only supports Kafka sinks.
+- Materialize currently only supports Avro formatted sinks that write to either a single partition topic or a Avro object container file.
+- On each restart, Materialize creates new, distinct topics and files for each sinks. Topic and file names are suffixed with {topic-name}-{sink global-id}-{startup-time}-{nonce}, and file names preserve the path argument's extension.
```
Maybe this would be better with an example?

ruchirK commented on doc/user/sql/create-sink.md:
````diff
@@ -82,6 +80,43 @@ FORMAT AVRO USING
     CONFLUENT SCHEMA REGISTRY 'http://localhost:8081';
 ```
 
+### Avro OCF sinks
````
Add an example here and for the Kafka sink case showing how to look up actual file/topic name

sploiselle commented on doc/user/sql/create-sink.md:
```diff
@@ -26,22 +26,28 @@ Field | Use
 **IF NOT EXISTS** | If specified, _do not_ generate an error if a sink of the same name already exists. <br/><br/>If _not_ specified, throw an error if a sink of the same name already exists. _(Default)_
 _sink&lowbar;name_ | A name for the sink. This name is only used within Materialize.
 _item&lowbar;name_ | The name of the source or view you want to send to the sink.
+**FILE** _path-prefix_ | The absolute path and file name prefix of file to create and write to.
```
nit: I realize that this is annoying, but can you amend the variable names to use underscores instead of hyphens to make them consistent with the other variable names? Glad to go back through and do this for you if it's too obnoxious.

sploiselle commented on doc/user/sql/create-sink.md:
```diff
@@ -26,22 +26,28 @@ Field | Use
 _sink&lowbar;name_ | A name for the sink. This name is only used within Materialize.
 _item&lowbar;name_ | The name of the source or view you want to send to the sink.
+**FILE** _path-prefix_ | The absolute path and file name prefix of file to create and write to.
+
```
Can we add a line for **AVRO OCF** here detailing what that option does?

ruchirK replied:
So I tried to rephrase the summary for path_prefix in a way that would make sense. Let me know if it actually does

sploiselle replied:
The term prefix is throwing me off here because it's an entire filename; it's not strictly a prefix. I think the term `path_prefix` here could be fine, though, if we're more verbose here. Something like:

> The absolute path and filename you want to create and write to. The filename will be modified to let Materialize create a unique file each time Materialize starts, but the file extension will not be modified. You can find more details !MAKE THIS A LINK! here !END LINK!

I'm not saying you must use that verbatim, but something akin to that would help someone understand what to expect based on what they enter, I think.

ruchirK replied:
Ok that makes sense I'll update

sploiselle commented on doc/user/sql/create-sink.md:
```diff
@@ -26,22 +26,28 @@ Field | Use
 
-- Materialize currently only supports Kafka sinks.
+- Materialize currently only supports Avro formatted sinks that write to either a single partition topic or a Avro object container file.
+- On each restart, Materialize creates new, distinct topics and files for each sinks. Topic and file names are suffixed with {topic-name}-{sink global-id}-{startup-time}-{nonce}, and file names preserve the given extension.
```
s/each sinks/each sink/

sploiselle replied:
wrap {topic-name}-{sink global-id}-{startup-time}-{nonce} in backticks for code formatting

sploiselle replied:
If that is what they're suffixed with, what exactly are they prefixed with? An example might make it easier to see what this does.

ruchirK replied:
I rephrased to hopefully make this clearer. Let me know if you think it needs more work

sploiselle commented on doc/user/sql/create-sink.md:
```diff
@@ -26,22 +26,28 @@ Field | Use
-- Materialize currently only supports Kafka sinks.
+- Materialize currently only supports Avro formatted sinks that write to either a single partition topic or a Avro object container file.
+- On each restart, Materialize creates new, distinct topics and files for each sinks. Topic and file names are suffixed with {topic-name}-{sink global-id}-{startup-time}-{nonce}, and file names preserve the given extension.
+- Materialize stores information about actual topic names and actual file names in the `mz_kafka_sinks` and `mz_avro_ocf_sinks` log sources.
```
Can I get you to link to the example here? I at first thought these would require a schema name but realize that isn't the case!

sploiselle commented on doc/user/sql/create-sink.md:
```diff
@@ -26,22 +26,28 @@ Field | Use
+- Materialize currently only supports Avro formatted sinks that write to either a single partition topic or a Avro object container file.
+- On each restart, Materialize creates new, distinct topics and files for each sinks. Topic and file names are suffixed with {topic-name}-{sink global-id}-{startup-time}-{nonce}, and file names preserve the given extension.
+- Materialize stores information about actual topic names and actual file names in the `mz_kafka_sinks` and `mz_avro_ocf_sinks` log sources.
+- Materialize generates Avro schemas for views and sources that are stored in sinks. The generated schemas have a Debezium style diff envelope to capture changes in the input view or source.
```
s/`Debezium style diff envelope`/`[Debezium-style diff envelope](docs/overview/api-components/#envelopes)`

ruchirK replied:
done

sploiselle commented on doc/user/sql/create-sink.md:
```diff
@@ -75,13 +81,78 @@ CREATE MATERIALIZED VIEW frank_quotes AS
+
+When creating Avro Object Container File (OCF) sinks, Materialize creates a new sink file using the naming scheme defined above and appends the Avro schema data in its header.
+
+## Examples
```
Can we restructure the examples so they don't collide? I have a slight preference for each doc to have one `Examples` section that's an `h2`, and then develop some other taxonomy from there, e.g. `### Kafka sinks`, `### Avro OCF sinks` and then make each example an `h4`. If you dislike that, can we amend the `Examples` text to make clear which type they're an example of?

ruchirK replied:
done I like that format better too!

sploiselle commented on doc/user/release-notes.md:
```diff
@@ -74,9 +74,15 @@ Use relative links (../path/to/doc), not absolute links
+
 - Support `EXPLAIN stage_name PLAN FOR ...` to display the plan at various stages of the planning process.
 
+- Support Avro Object Container File (OCF) [sinks](../sql/create-sink/#avro-ocf-sinks)
```
nit: period at end of line

ruchirK commented:
Ok -- I've incorporated your feedback and reworked slightly @sploiselle lmk what you think.
sidenote: theres a big lesson in this for me of "make features that are easily explainable for people"

sploiselle commented:
LGTM! Thanks for your work in going through this.
